### PR TITLE
Robustness

### PR DESCRIPTION
**Pull request by 2ni**
- made it more robust
- it seems that communication sometimes breaks when sending commands if we ignore the response
- cleanup up functions. always use public function command(...)
- added wiring of miromico module
- slightly updated examples

### CODE REVIEW

File: examples/03_send/03_send.ino [hunk 1]
```diff
@@ -19,10 +17,11 @@ void setup() {
   modem.join(appeui, appkey);
 
   unsigned long current_time = millis();
+  Serial.println("waiting ");
```
Comment (gonzalocasas) [inline examples/03_send/03_send.ino]: how about moving this waiting code to a function in `LoRaWANModem` (something like `wait_until_join()`) and then we could add a bit of a delay to not hammer the module with `getevent` requests non-stop?

File: src/loramodem.cpp [hunk 2]
```diff
@@ -42,10 +44,11 @@ bool LoRaWANModem::is_joining(void (*join_done)(Event_code code)) {
   uint8_t response[3] = {0};
   Status s = command(CMD_GETEVENT, response, &len);
   if (s != OK) {
-    Serial.printf(DBG_ERR("join event cmd error: 0x%02x") "\n", (uint8_t)s);
+    Serial.printf(DBG_ERR("joining event cmd error: 0x%02x") "\n", (uint8_t)s);
   }
 
   if (response[0] == EVT_JOINED || response[0] == EVT_JOINFAIL) {
```
Comment (gonzalocasas) [inline src/loramodem.cpp:54]: The `JOIN_FAIL` event indicates that one round of trying all channels and data rates failed, but the modem keeps on trying (and eventually will generate multiple `JOIN_FAILED` events), so the join is not done when this event occurs.

**Review by gonzalocasas — COMMENTED** [review]
With the update, I actually have a bit of a harder time joining. Check out this screenshot:
image
It shows it sent a join request many times, but somehow the module missed it and only managed to join many messages later.

There's also that very long message that always appear as first message after join which makes me think we're sending some random buffer at some point :P

File: examples/03_send/03_send.ino (hunk 1 above)
Comment (2ni) [reply examples/03_send/03_send.ino]: good idea! I actually did it the other way round:
rename the current join() to command_join() and have a join() which waits for the network. It'll return OK or FAIL.

About issues above: I assume these are the device logs. What does the gw say? For me it works with the modem nearby at least. Maybe we can have a look at it in a call?

Comment (gonzalocasas) [reply examples/03_send/03_send.ino]: Yep! Let's do that! I'll be free(er) a little later in the morning

**Review by gonzalocasas — APPROVED** [review]
yay!

**Pull request merged**